content-cache/tests/integration/conftest.py

```diff
@@ -70,17 +77,32 @@ async def deploy_applications_fixture(
     app_name: str,
     config_app_name: str,
     cert_app_name: str,
+    metric_app_name: str,
 ) -> AsyncIterator[dict[str, Application]]:
     """Deploy all applications in parallel."""
-    app_task = model.deploy(charm_file, app_name, base="ubuntu@24.04")
-    config_app_task = model.deploy(config_charm_file, config_app_name, num_units=0)
-    cert_app_task = model.deploy(
-        CERT_CHARM_NAME, cert_app_name, base="ubuntu@22.04", channel="latest/edge"
+    app_deploy = model.deploy(charm_file, app_name, base="ubuntu@24.04")
+    config_app_deploy = model.deploy(
+        config_charm_file, config_app_name, base="ubuntu@24.04", num_units=0
+    )
+    cert_app_deploy = model.deploy(
+        CERT_CHARM_NAME, cert_app_name, channel="latest/edge", base="ubuntu@22.04"
+    )
+    # The pinning to revision 319 due to a `model.deploy` issue. Ideally, the revision is not pinned.
+    # The `model.deploy` is unable to resolve to a workable revision, hence hardcoding to revision 319.
+    metric_app_deploy = model.deploy(
+        METRIC_CHARM_NAME, metric_app_name, channel="latest/edge", base="ubuntu@24.04", revision=319, num_units=0
+    )
```

Comment:
Just specifying the channel to `latest/edge` and base to `ubuntu@24.04` would result in unable to find a matching revision.
But it seems the revision 319 should fit the bill. Hence, I am hardcoding the revision for now.
<img width="1363" alt="Screenshot 2024-12-09 at 14 27 26" src="https://github.com/user-attachments/assets/a24a1326-12f2-43dc-b6f6-ef14327dba10">

Reply:
I think this might be a bug on `model.deploy`. 
Since running `juju deploy grafana-agent --channel latest/edge --base ubuntu@24.04` in shell does work.
I will submit a bug ticket.

Reply:
Filed a issue on GitHub: https://github.com/juju/python-libjuju/issues/1240